src/Services/DBusServer.vala

```diff
@@ -66,11 +66,29 @@ public class DBusServer : Object {
     public void uninstall (string component_id) throws Error {
         var client = AppCenterCore.Client.get_default ();
         var package = client.get_package_for_component_id (component_id);
+
         if (package == null) {
-            throw new IOError.FAILED ("Failed to find package for '%s' component ID".printf (component_id));
+            var error = new IOError.FAILED ("Failed to find package for '%s' component ID".printf (component_id));
+            new UninstallFailDialog (package, error).present ();
+            throw error;
+        }
+
+        var uninstall_confirm_dialog = new UninstallConfirmDialog (package);
+
+        if (uninstall_confirm_dialog.run () == Gtk.ResponseType.ACCEPT) {
+            package.uninstall.begin ((obj, res) => {
+                try {
+                    package.uninstall.end (res);
+                } catch (Error e) {
+                    // Disable error dialog for if user clicks cancel. Reason: Failed to obtain authentication
+                    if (!(e is Pk.ClientError) || e.code != Pk.ErrorEnum.NOT_AUTHORIZED + 0xFF) {
```

Comment:
Could you add a comment about the 0xFF? I wrote one in my previous comments that you could use.